### PR TITLE
feat(log-explorer): arrow key and deeper shortcuts for log explorer

### PR DESCRIPTION
**Pull request by awaseem**
Closes [FE-3378](https://linear.app/supabase/issue/FE-3378/featlogs-keyboard-shortcuts-for-function-logs-invocations-and-logs).

## Summary
Adds a shared shortcut registry for every `LogsPreviewer` surface — Function Logs, Function Invocations, and the Logs Explorer — and brings the grid keyboard model in line with the Auth Users / Table Editor patterns.

## Shortcuts

| Key | Action |
| --- | --- |
| `↑` / `↓` | Move single-row selection; opens side panel |
| `Shift+Space` | Toggle current row in multi-select |
| `Mod+A` | Toggle all visible rows in multi-select |
| `Esc` | Staged: clear multi-select → close side panel |
| `Shift+R` | Refresh logs |
| `Shift+H` | Toggle histogram |
| `Shift+L` | Load older logs |
| `Shift+P` | Open time range picker |
| `Mod+Shift+J / M / C` | Copy selected rows as JSON / Markdown / CSV (existing global handler) |

## Other changes
- `ShortcutTooltip` on search, refresh, histogram, load older, and time-picker controls.
- `onSearchInputEscape` wired on the logs search bar (clear → blur).
- Visual row highlight (`rdg-row--focused`) when a row is keyboard-focused or multi-selected.
- Multi-select copy dropdown gains a **Copy as CSV** entry and shows the keybind on each item via `ShortcutBadge`.
- Manual arrow-nav (`navigate()`) updates `selectedRow` directly without going through `onRowClick`, so multi-select checkmarks survive keyboard navigation.

## Test plan
- [x] Function Logs and Function Invocations: all shortcuts above fire while the page is mounted, no firing in other tabs.
- [x] Logs Explorer: same shortcuts work; copy keybinds still copy *all* rows when nothing is multi-selected.
- [x] Arrow keys on first load select the first row even when the focus sink is the active element.
- [x] Selecting rows via checkbox or `Shift+Space`, then pressing arrow keys, preserves the checkmarks.
- [x] Escape on a populated search input clears it; Escape on an empty input blurs it.
- [x] Esc with multi-select active clears the selection before closing the side panel.

## Summary by CodeRabbit

* **New Features**
  * CSV export for selected logs (adds CSV alongside JSON and Markdown).
  * New logs-preview keyboard shortcuts: search focus, refresh, chart toggle, date picker, load older, navigation, and selection.

* **Improvements**
  * Shortcut badges/tooltips integrated across the logs UI.
  * Search input focus/ref support and controllable date-picker open state.
  * Improved no-results/error displays and expanded copy dropdown sizing.

* **Tests**
  * Added CSV formatting tests covering RFC 4180 edge cases.

[Review Change Stack](https://app.coderabbit.ai/change-stack/supabase/supabase/pull/45989?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: apps/studio/components/interfaces/Settings/Logs/Logs.utils.ts [hunk 1]
```diff
@@ -745,6 +746,10 @@ export function formatLogsAsJson(rows: LogData[]): string {
   return JSON.stringify(rows, null, 2)
 }
 
+export function formatLogsAsCsv(rows: LogData[]): string {
+  return convertResultsToCSV(rows as unknown as Record<string, unknown>[]) ?? ''
+}
```
[the remark this answers is not in the document]
Comment (awaseem) [reply apps/studio/components/interfaces/Settings/Logs/Logs.utils.ts:751]: This should be fixed now

**Review by charislam — APPROVED** [review]

**Review by dnywh — APPROVED** [review]
Tested on local.
